src/puppetlabs/trapperkeeper/services/webserver/jetty10_websockets.clj

```diff
@@ -95,11 +104,18 @@
       (onWebSocketClose [statusCode ^String reason]
         (let [^WebSocketAdapter this this]
           (proxy-super onWebSocketClose statusCode reason))
+        (.countDown closureLatch)
         (on-close this statusCode reason))
       (onWebSocketBinary [^bytes payload offset len]
         (let [^WebSocketAdapter this this]
           (proxy-super onWebSocketBinary payload offset len))
         (on-bytes this payload offset len))
+      (awaitClosure []
+        (try
+          (when-not (.await closureLatch 30 TimeUnit/SECONDS)
+            (log/debug (i18n/trs "Timed out awaiting closure of websocket from remote.")))
+          (catch InterruptedException e
+            (log/debug (i18n/trs "Thread was interrupted when awaiting closure of websocket from remote. Exception: " e)))))
```

Comment:
This failure should be an unusual circumstance, so we want to log it at a higher level, and allow the `log/info` function to handle displaying the exception information itself by providing it as the first argument.  Is there any identifying information that would be helpful to include (request path maybe?)

```suggestion
            (log/info e (i18n/trs "Thread was interrupted when awaiting closure of websocket from remote.")))))
```

Reply:
Updated